### ISSUE TITLE
Using a JSON configuration for the Ruby linter

### ISSUE THREAD

**Issue by bvhme** [markdownlint/markdownlint#262]
Hey, I know this is maybe a weird question but my project is a Jekyll project based on Ruby and I'd love to use the `mdl` linter on Travis, however the configuration file I've got is the markdownlint js style `.markdownlint.json` coming from a VS Code contributor. 

Is there a way to, or a command to feed this to `mdl` or transform it to a file `mdl` can understand?

This would enable all the collaborators on their project to keep using the same tools whilst enjoying the same rules being linted by their editors.

**Comment by HT43-bqxFqB** [comment]
This would help us as well, with people using VSC or vim.

**Comment by robert-7** [comment]
Agreed. More information about supported formats for VS Code can be found [here under the Configure section](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint). Apparently, they include:

* Javascript
* JSON
* YAML

At the moment, I'm considering just hacking up a script to transform a YAML-formatted file to a Ruby formatted file 😬 . Example below:

```yaml
default: true
MD007:
  indent: 4
MD013: false
MD024: false
MD025: false
MD033: false
MD036: false
MD041: false
```

```ruby
all
rule 'MD007', :indent => 4
exclude_rule 'MD013'
exclude_rule 'MD024'
exclude_rule 'MD025'
exclude_rule 'MD033'
exclude_rule 'MD036'
exclude_rule 'MD041'
```

**Comment by jaymzh** [comment]
I think the JS port of markdownlint takes a JSON file, not sure. We certainly could, but having it be a ruby file means people can and do do stuff like:
```ruby
(5..8).each do ||x|
  exclude_rule "MD#{x}"
end
```
which obviously you can't do in JSON/YAML.

How about if you did the other way? Drop off a config file that reads in extra stuff from a JSON file. Ala:

```ruby
d = JSON.parse(File.read(.mdl_json))
d['excluded_rules'].each do |rule|
  exclude_rules rule
end
```

Then you can drop off JSON, YAML, INI, XML, or whatever else you want as long as your ruby install has a library for it

**Comment by mvz** [comment]
Maybe I'm missing something, but one nice thing about the json/yaml configuration is that it's one file, rather than a two-step `.mdlrc` + style file system.

**Comment by hlovdal** [comment]
The following is a minimal workaround, mdl should really add support for reading configuration from a `.markdownlint.json` file.

-------------


My initial `.markdownlint.json` is just 

```json
{
	"MD024": false
}
```

so I updated `Makefile` with a new rule to generate style file and add it as dependency to the lint rule:

```make
mdl-lint: .mdl_style.rb
	find ... \
	| xargs -0 --no-run-if-empty bundle exec mdl

.mdl_style.rb: .markdownlint.json
	ruby bin/mk_mdl_style.rb $^ $@

...

clean:
	rm -f .mdl_style.rb
```

The script is the following minimum:

```ruby
require 'json'

if ARGV.length != 2
	throw "This script should be called with two arguments, input .markdownlint.json file and output .mdl_style.rb file"
end

inputfile = ARGV[0]
outputfile = ARGV[1]

data_hash = JSON.parse(File.read(inputfile))

begin
	file = File.open(outputfile, "w")
	data_hash.each do |i|
		rule = i[0]
		value = i[1]
		#puts "rule: #{rule}"
		#puts "value: #{value}"
		if value == false
			text = "exclude_rule '#{rule}'\n"
			#puts text
			file.write(text)
		else
			throw "Unsupported value #{value} for rule #{rule}"
		end
	end
rescue IOError => e
	throw "IOError #{e}"
ensure
	file.close unless file.nil?
end
```

**Comment by hlovdal** [comment]
Except it does not work...

```shell
$ cat .mdlrc.rb 
all
exclude_rule 'MD024'
$ bundle exec mdl -c .mdlrc.rb CHANGELOG.md 
CHANGELOG.md:24: MD024 Multiple headers with the same content
$
```

Even when also throwing in an `all` like suggested earlier, mdl still complains about rule MD024 even though I've explicitly tried to disable it. Any hints on why?

Even using the excample config from this repository, e.g. `bundle exec mdl -c ../../github/markdownlint/.mdl_style.rb CHANGELOG.md` fails the same way.
